spec/outputs/graphite_spec.rb

```diff
@@ -1,86 +1,49 @@
 require_relative '../spec_helper'
 
-describe LogStash::Outputs::Graphite do
-
-  let(:port) { 4939 }
-  let(:config) do <<-CONFIG
-     input {
-        generator {
-          message => "foo=fancy bar=42"
-          count => 1
-          type => "generator"
-        }
-      }
-
-      filter {
-        kv { }
-      }
-
-      output {
-        graphite {
-          host => "localhost"
-          port => #{port}
-          metrics => [ "hurray.%{foo}", "%{bar}" ]
-        }
-      }
-  CONFIG
+class LogStash::Outputs::Graphite
+  def socket
+    @socket
   end
+  def connect
+    @socket = Mocks::Server.new
```

Comment:
I'm good with one way or another, I see I'm on the other side, trying to not do too much reopening :-) 

however I would say i recommend to move this class some other place, so it's not polluting the test definition. However it must be in a way that the redefinition is still clear that is happening.